internal/pkg/cli/svc_init.go

```diff
@@ -355,6 +362,15 @@ func (o *initSvcOpts) Ask() error {
 
 // Execute writes the service's manifest file and stores the service in SSM.
 func (o *initSvcOpts) Execute() error {
+	if !o.allowAppDowngrade {
```

Comment:
ok just thinking out loud 💭 what if we created a separate struct to embed in `initSvcOpts`, `initJobOpts`, and `initEnvOpts` like:
```
type downgradeValidator struct {
  allowDowngrade   bool
  newVersionGetter func() (versionGetter, error)
}

func (d *downgradeValidator) validate() error {}
```

then in `Execute()`
```
if err := o.appDowngradeValidator.validate(); err != nil {
}
```

and in `newOpts()`
```
o.downgradeValidator.newVersionGetter = func() (versionGetter, error) {
  return describe.NewAppDescriber(o.appName)
}
```

Reply:
trying to think of something we can reuse when we get to `svc deploy` and `env deploy`

Reply:
Hmm i thought about it but would prefer leaving `allowAppDowngrade` out because it's controlled by a flag. Always calling `o.appDowngradeValidator.validate()` makes less sense to me.

> trying to think of something we can reuse when we get to svc deploy and env deploy

Maybe it's ok? Within the if statement we just init the struct and call validate method.

Reply:
we could use the embedded value as a flag?
```
cmd.Flags().BoolVar(&vars.appDowngradeValidator.allowDowngrade, allowDowngradeFlag, false, allowDowngradeFlagDescription)
```

but yeah, you're right, it's not necessarily a ton of logic. i'll leave it up to you!